specifications/M1/credential.issuance.protocol.md

```diff
@@ -0,0 +1,313 @@
+Token to provide the pre-authorization code to the issuer.
+
+### 3.1.1. Credential Request Parameters
+
+The Credential Request `POST` body MUST be a JSON object with the following properties:
+
+- `format`: REQUIRED. A JSON string that describes the format of the credential to be issued. Implementations MUST
+  support the `ldp_vc` format as defined by
+  the [OpenID for Verifiable Credential Issuance specification](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-e.1.3).
+  Implementations MAY support other VC formats.
+- `types`: REQUIRED. A JSON array of strings that specifies the VC type being requested.
+
+The following is a non-normative example of a credential request:
+
+```
+POST /credential HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Authorization: Bearer ......
+
+{
+   "format":"ldp_vd",
+   "types":[
+      "VerifiableCredential",
+      "EntityCredential"
+   ],
+}
+```
+
+On successful receipt of the request, the Credential Issuer MUST respond with an `201 CREATED` with the `Location`
+header set to the location of the request status in the [Credential Request Status](#credential-request-status-endpoint)
+section.
+
+The issuer MAY respond with `401 Not Authorized` if the request is unauthorized or other `HTTP` status codes to indicate
+an exception.
+
+If the VC request is approved, the issuer will respond with a write-request to the client's `Credential Service` using
+the Storage API defined in the [Verifiable Presentation Protocol](./verifiable.presentation.protocol.md#5-storage-api).
+
+# 4. Credential Offer Flow
+
+Some scenarios involve the Credential Issuer making an initial offer. For example, an out-of-band may result in a
+credential offer. Or, a Credential Issuer may start a key rotation process which involves sending updated credentials to
+holders signed with a new key. In this case, the issuer can proactively prompt holders to request a new credential
+during the key rotation period.
+
+## 4.1. Credential Offer Endpoint
+
+Communication with the `Credential Offer Endpoint` MUST utilize TLS.
+
+The credential offer endpoint MUST be available under the `POST` method at `/offers` relative to the base URL of the
+holder's `Credential Service` base URL. Issuer's can obtain this URL by resolving the holder's DID and inspecting
+its `CredentialService` service entry.
+
+### 4.1.1. Credential Offer Parameters
+
+The Credential Offer `POST` body MUST be a JSON object with the following properties:
+
+- `credential_issuer`: REQUIRED. The URL of the Credential Issuer, the `Credential Service` is requested to obtain one
+  or more credentials from.
+- `credentials`: REQUIRED. A JSON array, where every entry is a JSON object or a JSON string. If the entry is an object,
+  its data MUST adhere to [the Credentials Object Parameters](#the-credentials-object-parameters). If the entry is a
```

Comment:
Use a sub-list to define rules, instead of using "if... else..."

Reply:
Not sure I follow. We could number these but the steps are not strict order

Reply:
```suggestion
- `credentials`: REQUIRED. ...
    - entry type object: data MUST adhere to [the Credentials Object Parameters](#the-credentials-object-parameters)
    - entry type string: value MUST be one of the id values in one of the objects in the `credentials_supported`
```

along these lines